pythran/toolchain.py

```diff
@@ -169,8 +169,10 @@ def __str__(self):
                             )
         mod.add_to_includes(*map(Include, _extract_specs_dependencies(specs)))
         mod.add_to_includes(*content.body)
+
+        # the lambda is here to harmonize inconsistent return type between py2 and py3
         mod.add_to_init(
-            Line('#ifdef PYTHONIC_TYPES_NDARRAY_HPP\nimport_array()\n#endif'))
+            Line('#ifdef PYTHONIC_TYPES_NDARRAY_HPP\n[](){import_array(); return NUMPY_IMPORT_ARRAY_RETVAL;}();\n#endif'))
```

Comment:
On Wed, Sep 02, 2015 at 11:54:53AM -0700, pbrunet wrote:
> >          mod.add_to_init(
> > -            Line('#ifdef PYTHONIC_TYPES_NDARRAY_HPP\nimport_array()\n#endif'))
> > +            Line('#ifdef PYTHONIC_TYPES_NDARRAY_HPP\n[](){import_array(); return NUMPY_IMPORT_ARRAY_RETVAL;}();\n#endif'))
> 
> can't you just add a void cast before to say we ignore the result?

No, because import_array is a macro